dbms/src/Common/MPMCQueue.h

```diff
@@ -289,26 +304,37 @@ class MPMCQueue
                 return read_pos < write_pos || !isNormal();
             };
             if (!wait(lock, reader_head, node, pred, deadline))
-                return Result::TIMEOUT;
+                is_timeout = true;
         }
         if (!isCancelled() && read_pos < write_pos)
```

Comment:
add comment same as `assignObj`
```suggestion
        /// double check status after potential wait
        if (!isCancelled() && read_pos < write_pos)
```